packages/block-editor/src/components/page-template-picker/use-page-template-picker.native.js

```diff
@@ -12,18 +12,25 @@ export const __experimentalUsePageTemplatePickerAvailable = () => {
 };
 
 export const __experimentalUsePageTemplatePickerVisible = () => {
-	const isTemplatePickerAvailable = __experimentalUsePageTemplatePickerAvailable();
+	return useSelect(
+		( select ) => {
+			const { getBlockOrder, getBlock } = select( 'core/block-editor' );
 
-	return useSelect( ( select ) => {
-		const { getBlockOrder, getBlock } = select( 'core/block-editor' );
+			const blocks = getBlockOrder();
+			const isEmptyBlockList = blocks.length === 0;
+			const firstBlock = ! isEmptyBlockList && getBlock( blocks[ 0 ] );
 
-		const blocks = getBlockOrder();
-		const isEmptyBlockList = blocks.length === 0;
-		const firstBlock = ! isEmptyBlockList && getBlock( blocks[ 0 ] );
-		const isOnlyUnmodifiedDefault =
-			blocks.length === 1 && isUnmodifiedDefaultBlock( firstBlock );
-		const isEmptyContent = isEmptyBlockList || isOnlyUnmodifiedDefault;
+			const isOnlyUnmodifiedDefault =
+				blocks.length === 1 && isUnmodifiedDefaultBlock( firstBlock );
 
-		return isEmptyContent && isTemplatePickerAvailable;
-	}, [] );
+			const isEmptyContent = isEmptyBlockList || isOnlyUnmodifiedDefault;
+			const isTemplatePickerAvailable = __experimentalUsePageTemplatePickerAvailable();
+
+			return isEmptyContent && isTemplatePickerAvailable;
+		},
+		[
+			isUnmodifiedDefaultBlock,
+			__experimentalUsePageTemplatePickerAvailable,
```

Comment:
Thinking about this more, since these are external functions, and they'll always be the same functions, I can remove them from the dependency array. Am I correct?

Reply:
`__experimentalUsePageTemplatePickerAvailable`

This reads like a hook, I don't think we should be using it "inside" useSelect, we may break the hooks rules that way.

Reply:
Thanks again for catching that. I've moved that out of the `useSelect` call and changed the dependency array to include `isTemplatePickerAvailable`. I've also removed `isUnmodifiedDefaultBlock` from the dependency array. That should be fine, right?